src/main/java/com/mycompany/mszczepienia/service/VisitService.java

```diff
@@ -112,6 +115,21 @@ public void updateMissedVisitsStatus() {
         });
     }
 
+    @Transactional
+    public VisitDto cancelVisit(Long visitId){
+        System.out.println(visitId);
+        Visit visit = visitRepository.getById(visitId);
```

Comment:
``
visitRepository.getById(visitId);
``
Repository method getBy... doesn't hit the database, it just returns proxy reference useful for setters only (without reading). Calling getters on this proxy is not allowed. Replace "get" with "find" to get the entity in a proper way.